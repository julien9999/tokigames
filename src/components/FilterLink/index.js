import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

export default function ({ action, type, children, property }) {
  const active = useSelector(state => type === state[property]);
  const dispatch = useDispatch();

  return (
    <li className="nav-item">
        <a 
          onClick={() => dispatch(action(type))}
          className={`nav-link ${active ? 'active' : null}`} 
        >
          {children}
        </a>
    </li>
  )
}