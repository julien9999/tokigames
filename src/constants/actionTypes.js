export const INITIAL_STATE = 'INITIAL_STATE';

/**
 * Content action types
 */
export const CONTENT_UPDATE_SUCCESS = 'CONTENT_UPDATE_SUCCESS';
export const CONTENT_UPDATE_ERROR = 'CONTENT_UPDATE_ERROR';
export const ADD_FLIGHT = 'ADD_FLIGHT';
export const DELETE_FLIGHT = 'DELETE_FLIGHT';


/**
 * Order action types
 */
export const SET_ORDER = 'SET_ORDER';
export const SHOW_DURATION = 'SHOW_DURATION';
export const SHOW_STARTTIME = 'SHOW_STARTTIME';
export const SHOW_ENDTIME = 'SHOW_ENDTIME';

/**
 * Filter action types
 */
export const SHOW_ALL_FILTER = 'SHOW_ALL_FILTER';
export const SET_FILTER = 'SET_FILTER';
export const SHOW_CHEAP = 'SHOW_CHEAP';
export const SHOW_BUSINESS = 'SHOW_BUSINESS';
