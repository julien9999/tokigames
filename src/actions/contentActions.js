import moment from 'moment';
import { getFlightsCheap, getFlightsBusiness } from '../api/content';
import * as types from '../constants/actionTypes';

function getContentSuccess(payload) {
  return { type: types.CONTENT_UPDATE_SUCCESS, payload };
}

function getContentError() {
  return { type: types.CONTENT_UPDATE_ERROR };
}

function addContentSuccess(flight) {
  return { type: types.ADD_FLIGHT, flight };
}

function deleteContentSuccess(id) {
  return { type: types.DELETE_FLIGHT, id };
}

/**
 * Load Content
 *
 * Get content
 */
export const loadContents = () => {
  return (dispatch) => {
    return Promise.all([getFlightsCheap(), getFlightsBusiness()]).then((response) => {
      const data = [
        ...response[0].data.data.map((element) => ({ 
          ...element, 
          duration: +(element.arrival - element.departure),
          durationMinutes: moment(new Date(element.departure * 1000)).diff(moment(new Date(element.arrival * 1000)), 'minutes'),
          startDate: moment(new Date(element.departure * 1000)).format('MM/DD/YYYY'),
          endDate: moment(new Date(element.arrival * 1000)).format('MM/DD/YYYY'),
          type: 'cheap',
        })),
        ...response[1].data.data.map((element) => ({ 
          startDate: moment(new Date(element.departureTime * 1000)).format('MM/DD/YYYY'),
          endDate: moment(new Date(element.arrivalTime * 1000)).format('MM/DD/YYYY'),
          duration: +(element.arrivalTime - element.departureTime),
          durationMinutes: moment(new Date(element.arrivalTime * 1000)).diff(moment(new Date(element.departureTime * 1000)), 'minutes'),
          arrival: element.arrivalTime,
          departure: element.departureTime,
          route: `${element.arrival}-${element.departure}`,
          type: 'business' 
        }))
      ];

      const indexedData = data.map((element, index) => {
        return {
          ...element,
          index
        }
      })

      dispatch(getContentSuccess(indexedData));
    })
    .catch(err => {
      dispatch(getContentError());
    });
  };
};

export const addFlight = ({ arrival, departure, from, to, type }) => {
  return (dispatch) => {
    dispatch(addContentSuccess({
      type: types.ADD_FLIGHT,
      arrival: moment(arrival, 'YYYY-MM-DD').unix(),
      departure: moment(departure, 'YYYY-MM-DD').unix(),
      duration: moment(arrival, 'YYYY-MM-DD').unix() - moment(departure, 'YYYY-MM-DD').unix(),
      durationMinutes: moment(arrival, 'YYYY-MM-DD').diff(moment(departure, 'YYYY-MM-DD'), 'minutes'),
      startDate: moment(departure, 'YYYY-MM-DD').format('MM/DD/YYYY'),
      endDate: moment(arrival, 'YYYY-MM-DD').format('MM/DD/YYYY'),
      route: `${from}-${to}`,
      type: type ? type : 'business',
    }));
  }
}


export const deleteFlight = (id) => {
  return (dispatch) => {
    dispatch(deleteContentSuccess(id));
  }
}