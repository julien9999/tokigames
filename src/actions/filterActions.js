import * as types from '../constants/actionTypes';

export const setFilter = filter => ({
    type: types.SET_FILTER,
    filter
})