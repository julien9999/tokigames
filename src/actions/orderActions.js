import * as types from '../constants/actionTypes';

export const setOrder = order => ({
    type: types.SET_ORDER,
    order
})