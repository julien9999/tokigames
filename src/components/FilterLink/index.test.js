import React from 'react';
import renderer from 'react-test-renderer';
import FilterLink from './index';
import { setOrder } from '../../actions/orderActions';
import { SHOW_DURATION } from '../../constants/actionTypes';

jest.mock('react-redux', () => ({
  useDispatch: ()=> {},
  useSelector: () => true
}));

describe('<FilterLink>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <FilterLink 
            action={setOrder}
            type={SHOW_DURATION}
            children={"Duration"}
            property="order"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});