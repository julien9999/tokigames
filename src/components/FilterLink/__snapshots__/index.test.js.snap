// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterLink> renders correctly 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link active"
    onClick={[Function]}
  >
    Duration
  </a>
</li>
`;
